chart-parts/ingress.yaml

```diff
@@ -1,40 +1,39 @@
-{{- if .Values.services.ingress -}}
+{{- if .Values.ingress.enabled -}}
 ---
-# The cert and the key cannot be filled automatically. If we could it
-# would roughly look like the lines below. As is it is the
-# responsibility of the operator to retrieve the values of these
-# secrets from the deployed UAA and update this secret with the actual
-# values. And the responsibility of the developer to supply the proper
-# instructions to the operator.
-##
-# tls.crt: {{ default "" .Values.secrets.UAA_SERVER_CERT     | b64enc | quote }}
-# tls.key: {{ default "" .Values.secrets.UAA_SERVER_CERT_KEY | b64enc | quote }}
-
 apiVersion: "v1"
 kind: "Secret"
 type: kubernetes.io/tls
 metadata:
-  name: "{{ .Values.services.ingress.class }}-ingress-tls"
-  namespace: "{{ .Release.Namespace }}"
+  name: "ingress-tls"
+  namespace: {{ .Release.Namespace | quote }}
 data:
-  tls.crt: ""
-  tls.key: ""
+  tls.crt: {{ .Values.ingress.tls.crt | default "" | b64enc | quote }}
+  tls.key: {{ .Values.ingress.tls.key | default "" | b64enc | quote }}
 ---
 apiVersion: "extensions/v1beta1"
 kind: "Ingress"
 metadata:
-  name: "{{ .Release.Name }}-{{ .Values.services.ingress.class }}"
-  namespace: "{{ .Release.Namespace }}"
+  name: {{ .Release.Name | quote }}
+  namespace: {{ .Release.Namespace | quote }}
   annotations:
-    kubernetes.io/ingress.class: {{ .Values.services.ingress.class }}
-    {{ if eq .Values.services.ingress.class "nginx" -}}
+    {{- if empty (index .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+    kubernetes.io/ingress.class: "nginx"
+    {{- end }}
+    {{- if empty (index .Values.ingress.annotations "nginx.ingress.kubernetes.io/secure-backends") }}
     nginx.ingress.kubernetes.io/secure-backends: "true"
+    {{- end }}
+    {{- if empty (index .Values.ingress.annotations "nginx.ingress.kubernetes.io/backend-protocol") }}
     nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
-    nginx.ingress.kubernetes.io/ssl-redirect: "false" # Doesn't enforce HTTPS.
     {{- end }}
+    {{- if empty (index .Values.ingress.annotations "nginx.ingress.kubernetes.io/ssl-redirect") }}
+    nginx.ingress.kubernetes.io/ssl-redirect: "false"
+    {{- end }}
+{{- if not (empty .Values.ingress.annotations)}}
+{{ toYaml .Values.ingress.annotations | indent 4 }}
```

Comment:
Exploring alternative (but I don't even know for sure if this works):

```gotmpl
    {{- if hasKey .Values.ingress.annotations "kubernetes.io/ingress.class" | not -}}
        {{ $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" "ingress" }}
    {{- end }}
    {{- if hasKey .Values.ingress.annotations "nginx.ingress.kubernetes.io/secure-backends" | not -}}
        {{ $_ := set .Values.ingress.annotations "nginx.ingress.kubernetes.io/secure-backends" "true" }}
    {{- end }}
    {{- if hasKey .Values.ingress.annotations "nginx.ingress.kubernetes.io/backend-protocol" | not -}}
        {{ $_ := set .Values.ingress.annotations "nginx.ingress.kubernetes.io/backend-protocol" "HTTPS" }}
    {{- end }}
    {{- if hasKey .Values.ingress.annotations "nginx.ingress.kubernetes.io/ssl-redirect" | not -}}
        {{ $_ := set .Values.ingress.annotations "nginx.ingress.kubernetes.io/ssl-redirect" "false" }}
    {{- end }}
{{ toYaml .Values.ingress.annotations | indent 4 }}
```
Probably can stuff the `if hasKey / set / end` chunk in a helper function too…

Reply:
Sounds like a good improvement. I'll create a card for exploring it.

Reply:
I'm not particularly happy about pushing an improvement _suggested in the PR for the original code_ being deferred…

Reply:
Your comment was "Exploring alternative (but I don't even know for sure if this works)". I don't consider it to be expected to be addressed in the current PR. I'll do it now, though.

Reply:
Done.